src/QsCompiler/TextProcessor/QsExpressionParsing.fs

```diff
@@ -26,8 +26,10 @@ let internal unknownExpr = (InvalidExpr, Null) |> QsExpression.New
 /// For an expression of the given kind that is built from a left and right hand expression with the given ranges,
 /// builds the corresponding expression with its range set to the combined range. 
 /// If either one of given ranges is Null, builds an invalid expression with its range set to Null. 
-let private buildCombinedExpr kind (lRange, rRange) = 
-    QsPositionInfo.WithCombinedRange (lRange, rRange) kind InvalidExpr |> QsExpression.New // *needs* to be an invalid expression if the combined range is Null!
+let private buildCombinedExpr kind (lRange, rRange) =
+    match QsNullable.Map2 Range.Span lRange rRange with
+    | Value range -> { Expression = kind; Range = Value range }
+    | Null -> { Expression = InvalidExpr; Range = Null }
```

Comment:
It would be good to preserve the comment above.

Reply:
Resolved in #579.